common/utils/random.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/namer"
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type RandomGenerator struct {
+	ctx      *pulumi.Context
+	namer    namer.Namer
+	provider *random.Provider
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Where is this provider used ?

Reply:
Ah yes. It should be added as a dependent resource in `NewRandomString`